synapseclient/models/protocols/folder_protocol.py

```diff
@@ -0,0 +1,132 @@
+"""Protocol for the specific methods of this class that have synchronous counterparts
+generated at runtime."""
+
+from typing import Optional, TYPE_CHECKING, Protocol, Union, List
+from synapseclient import Synapse
+from synapseclient.models.services.storable_entity_components import (
+    FailureStrategy,
+)
+
+if TYPE_CHECKING:
+    from synapseclient.models import Project, Folder
+
+
+class FolderSynchronousProtocol(Protocol):
+    """
+    The protocol for methods that are asynchronous but also
+    have a synchronous counterpart that may also be called.
+    """
+
+    def store(
+        self,
+        parent: Optional[Union["Folder", "Project"]] = None,
+        failure_strategy: FailureStrategy = FailureStrategy.LOG_EXCEPTION,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "Folder":
+        """Storing folder and files to synapse.
```

Comment:
The File model has its own `store` procedure in `file_protocol`, so can this be reworded to mention only Folders

Reply:
True - However, if you have files in the `files` attribute or folders in the `folders` it will store those as well:
https://github.com/Sage-Bionetworks/synapsePythonClient/blob/685c3641d8e1b31a6fe99da67cd1086c17c2faf1/synapseclient/models/services/storable_entity_components.py#L66

Reply:
I added a more explicit comment here: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1068/commits/d8ca50963ce3839d83982f30696ff69a652b8e69